### ISSUE TITLE
Gnome shell memory leak

### ISSUE THREAD

**Issue by Cervoxx** [pop-os/gnome-shell#12]
My gnome shell has a memory leak, in the sense that its ram usage will continuously rise more and more as I use my system throughout the day. I've observed via the system monitor gnome shell using over 1GiB at times. This issue occurs on both my laptop and my desktop systems, albiet the issue appears to be more pronounced on my laptop. I cannot provide any steps to reproduce the issue since it seems to occur as soon as I turn on the computer. Restarting gnome shell brings the ram usage back down to regular levels, but it will slowly rise again. 

Laptop: Dell G3 3579

Pop!_OS 19.10 upgraded from 19.04.

This issue is not caused by any gnome extensions, I had disabled all the extensions on my computer and still observed the shell slowly rise in ram usage. 

The issue also occurs when the laptop lid is closed.

**Comment by mmstick** [comment]
Does it continue to rise above 1G, or does it remain in that region?

**Comment by Cervoxx** [comment]
Yes, I don't have screenshot evidence but I've seen it go above 1G.

**Comment by Cervoxx** [comment]
Speaking of photographic evidence, heres an example of it going over 1G. 
![image](https://user-images.githubusercontent.com/12489208/68350571-69fe0080-00c6-11ea-926a-6621063d232f.png)
Also note that right now I'm experiencing some stuttering.

**Comment by ksze** [comment]
There is also a ticket on the GNOME bug tracker, with perhaps more info: https://gitlab.gnome.org/GNOME/gnome-shell/issues/829

Interestingly, for some people, the memory leak happens with the `gnome-shell` instance run as the logged-in user; for some other people, the memory leak happens with the `gnome-shell` instance run as the `gdm` user, which is responsible for the log-in screen.

**Comment by daili0802** [comment]
I am experiencing the same issue with freshly installed pop_os 19.10, however, it did not happen while I was on 18.04. I seem to have gnome shell taking up 12gb / 16gb total, and ran as gdm.

**Comment by sawyermade** [comment]
I am getting this on 18.04 LTS, its real bad. Almost to the point where my system has become unusable. Mine will easily shoot up to 4GB in a few hours. Have to alt+f2 then r all the fucking time. I am about to switch to a different desktop or distribution.

**Comment by mdeneen** [comment]
Mine was at 7GB as the gdm user, and "alt + f2, r" did nothing.  This is on 19.10.

Are any of you using gnome-shell-extension-appindicator ?

**Comment by sawyermade** [comment]
I'm on 18.04 and did have the app indicator but disabling it didn't fix the prob for me. I actually turned off all extensions and it still jumped to an insane amount of memory :(

**Comment by daili0802** [comment]
I have since moved down to 18.04 from 19.10, and I don't have this issue any more. I did have the app indicator on before I believe.

**Comment by mdeneen** [comment]
I still have it on 19.10, but I've been using the system a lot and it was up for 8 days before I restarted it.  I've had other desktop environments up for months at a time in the past, though, and a little over a week is really nothing.

**Comment by tvjg** [comment]
I was seeing something like this on 19.10 too. It would grow unbounded until bad things started to occur. Uptime was usually about 3 days before I would force a restart. This is using a new-ish laptop with intel/nvidia hybrid graphics. I swapped from intel to nvidia graphics and the leak seems to have stopped. Is this a common thread for anyone else?

I've had a number of issues with sound and suspend recently, so it may just be lagging support for my hardware.

**Comment by sawyermade** [comment]
Yes, I am on desktop though, only discrete GPU Nvidia on 18.04 and it will
run through the roof on memory, my max has been 7gb+ and everything starts
lagging real bad. Alt+f2 then r seems to work so I don't have to restart,
but if I don't watch out and forget to do it, gnome shell will crash back
to the login screen. Real fucking annoying if you ask me.

**Comment by mdeneen** [comment]
I'm on 20.04 now with Intel graphics and it's using 4.8GB after 7 days.  I
put 32 GB into the machine, but I'd rather not waste it on the GUI.
The memory is in use by the gdm user.

**Comment by sawyermade** [comment]
Jesus Christ, that is literally insane. Shouldn't be more than 500-600MB,
800 tops. Why does this seem to be such a persistent problem? It isn't new
lol.
